File: java/src/main/java/present/Neighborhood.java
package present;

/**
 * A neighborhood comprised of a name and location
 */
public class Neighborhood {

  private final String name;
  private final Location location;

  /**
   * Constructs a new neighborhood
   *
   * @param name of the neighborhood
   * @param location of the neighborhood
   */
  public Neighborhood(String name, Location location) {
    this.name = name;
    this.location = location;
  }

  /** Returns the name of this neighborhood. */
  public String name() {
    return name;
  }

  /** Returns the location of this neighborhood. */
  public Location location() {
    return location;
  }
}

File: swift/nearby-neighborhoods/nearby-neighborhoods/present/Neighborhood.swift

/**
  A neighborhood comprised of a name and location
*/
public class Neighborhood {

  public let name: String
  public let location: Location

  /**
   Constructs a new neighborhood
    - parameter name: of the neighborhood
    - parameter location: of the neighborhood
  */
  public init(name: String, location: Location) {
    self.name = name
    self.location = location
  }
}

extension Neighborhood : Hashable {
  public var hashValue: Int {
    return name.hashValue &* location.hashValue
  }
  
  public static func == (lhs: Neighborhood, rhs: Neighborhood) -> Bool {
    return lhs.name == rhs.name && lhs.location == rhs.location
  }
}
